client-ui/src/app/components/lineage/lineage-graph/lineage-graph.component.ts

```diff
@@ -58,25 +57,13 @@ export class LineageGraphComponent implements OnInit, OnChanges, AfterViewInit {
   @ViewChild(CytoscapeNgLibComponent, {static: true})
   private cytograph: CytoscapeNgLibComponent
 
-  public ngOnInit(): void {
-    const writeNode = this.graph.nodes.find(n => n.data._type == 'Write')
-    writeNode.data.icon = operationIconCodes.get(OperationType.Write)
-    writeNode.data.color = operationColorCodes.get(OperationType.Write)
-  }
-
   public ngOnChanges(changes: SimpleChanges): void {
+    if (changes['graph']) this.refreshGraph()
```

Comment:
I guess, you do not want to do that on the very first change as far aa you are doing that in th ngAfterInit then one more time. 

So it should be:

`
if (changes['graph'] && !changes['graph'].isFirstChange()) {
      this.refreshGraph()
    }
`

Reply:
So I guess I need to do it for all three `if`s